compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/generators/AnnotationGenerator.kt

```diff
@@ -48,7 +49,8 @@ internal class AnnotationGenerator(private val components: Fir2IrComponents) : F
         irProperty.annotations +=
             property.annotations
                 .filter {
-                    it.useSiteTarget == null || it.useSiteTarget == AnnotationUseSiteTarget.PROPERTY
+                    !it.isJvmFieldAnnotation &&
```

Comment:
This `isJvmFieldAnnotation` check, as well as the next one, looks too specific for me. Normally we should check `@Target` of an annotation class, and if it's `FIELD`, then annotation with `null` use-site target isn't applicable to a property but applicable to its field.

Reply:
Just checked in the debugger: you can take `it.annotationTypeRef`, get its related `FirRegularClass`, and this class has `@Target` annotation. So let's do it in a generalized way :).

Reply:
Cite from Kotlin reference just to be precise:

```
If you don't specify a use-site target, the target is chosen according to the @Target annotation of the annotation being used. If there are multiple applicable targets, the first applicable target from the following list is used:

- param;
- property;
- field.
```

Reply:
Thanks for the suggestion!

I saw you already made some utils to retrieve meta annotation, like `@Retention`, so I migrated all the annotation-related utils into one place; extended it to retrieve `@Target` (a bit different, since it has array value); and then generalize property-v.s.-field targeting with targets in `@Target`. Now I can unmute one more. :)